src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationTaggerProvider.cs

```diff
@@ -56,13 +56,33 @@ internal partial class SemanticClassificationTaggerProvider : AsynchronousTagger
             _typeMap = typeMap;
         }
 
-        protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
+        protected override ITaggerEventSource CreateEventSource(ITextView textView, ITextBuffer subjectBuffer)
         {
+            this.AssertIsForeground();
+            const TaggerDelay Delay = TaggerDelay.Short;
+
             // Note: we don't listen for OnTextChanged.  Text changes to this this buffer will get
             // reported by OnSemanticChanged.
             return TaggerEventSources.Compose(
-                TaggerEventSources.OnSemanticChanged(subjectBuffer, TaggerDelay.Short, _semanticChangeNotificationService),
-                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer, TaggerDelay.Short));
+                TaggerEventSources.OnViewSpanChanged(textView, Delay),
+                TaggerEventSources.OnSemanticChanged(subjectBuffer, Delay, _semanticChangeNotificationService),
+                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer, Delay));
+        }
+
+        protected override IEnumerable<SnapshotSpan> GetSpansToTag(ITextView textView, ITextBuffer subjectBuffer)
+        {
+            this.AssertIsForeground();
```

Comment:
1. Collapsed regions work fine.  They will just mean we have a larger 'visible span'.  But in the worst case, we end up just having to reclassify the whole file (which is no worse than where we are today).

2. If you have the ```scroll map``` feature turned on, then it will *not* have semantic classifications in it for the region outside what is in view.  For TS we decided that was absolutely fine with us.  ```scroll map``` is an area that can have a slightly degraded experience when we're providing such a benefit to the mainline case of just editing.

Note that syntactic/lexical classification will still continue running inside ```scroll map```.  So most things will look good (and things like class declarations will still be teal).